### PR TITLE
Project cleanup

### PR DESCRIPTION
**Pull request by kageiit**
- Updated gradle/dependencies/plugins/buildtools to latest
- Simplified some classes by using groovier syntax
- Removed unnecessary helpers
- Simplified some existing helpers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,9 +174,8 @@ Maybe there are more caveats waiting for you, but for the super fast build broug
 The rest modules in this repo is a full example usage of OkBuck and BUCK.
 
 ## Compatibility
-OkBuck is tested under `gradle` 2.2.1 ~ 2.10, and `com.android.tools.build:gradle` 1.5.0. 
-Other versions will not considered compatible, if you have compatibility problem, please 
-fire an issue.
+OkBuck is tested under `gradle` 2.12, and `com.android.tools.build:gradle` 2.0.0.
+If other versions have compatibility problems, please file an issue.
```
Comment (Piasy) [inline README.md]: We need test compatibility.

Comment (kageiit) [inline README.md]: I tested it on a moderately large app using gradle 2.12 and android gradle plugin 2.0.0

I also updated all the sample/example projects to use these versions to verify

Comment (Piasy) [inline README.md]: I mean compatibility on older gradle version and android-gradle-plugin version, we should not force developer use the latest version.

Comment (kageiit) [inline README.md]: Of course. I can change the note. This is not forcing it. It still works with versions 1.5.0 to 2.0.0 of gradle android plugin

and 2.2.1 to 2.12 of gradle version

**Comment by kageiit** [comment]
@Piasy The travis check seems to fail compiling groovy for this PR.

It is weird because the same travis command I run locally passes. Can you just check what you are seeing on your machine if possible? I will try to debug in the meanwhile

**Comment by Piasy** [comment]
No problem.

**Comment by Piasy** [comment]
@kageiit My local build also succeed.

File: README.md (hunk 1 above)
Comment (Piasy) [inline README.md]: Great!

**Comment by kageiit** [comment]
Probably. I have updated the travis script to use jdk8

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-all.zip
```
Comment (Piasy) [inline gradle/wrapper/gradle-wrapper.properties]: We also need update gradle version in `buildSrc/gradle/wrapper/gradle-wrapper.properties`.

Comment (kageiit) [inline gradle/wrapper/gradle-wrapper.properties]: interesting. ok will change it

**Comment by Piasy** [comment]
@kageiit Maybe we should refer `DependencyType` in subclass of `Dependency` by `Dependency.DependencyType` ?

**Comment by kageiit** [comment]
Ok trying that

**Comment by Piasy** [comment]
And [this line](https://github.com/Piasy/OkBuck/blob/master/buildSrc%2Fbuild.gradle#L73):
```gradle
compile 'com.android.tools.build:gradle:1.5.0'
```

We should update to 2.0.0 too.

**Comment by kageiit** [comment]
yep its already updated as part of the PR

**Comment by Piasy** [comment]
@kageiit I think the root cause is the way we refer to `DependencyType`. We should update reference in all subclass of `Dependency`, there are 4.

**Comment by kageiit** [comment]
I think we need to only update in 2 places i.e where we do not inherit from Dependency and inherit from FileDependency instead. I have tried in my latest commit.

**Comment by kageiit** [comment]
Atleast it compiles groovy now. Travis box is unable to find build tools version 23.0.3 now :)

**Comment by kageiit** [comment]
Seems like a known issue with 23.0.3 https://github.com/travis-ci/travis-ci/issues/5891
I will revert just that part and update it once travis is ready. but it works locally with 23.0.3

**Comment by kageiit** [comment]
Finally green! Feel free to merge. I will open a PR for cleaning up the duplicate detection next :)

**Comment by Piasy** [comment]
Great job!

**Pull request merged**